src/Record.ts

```diff
@@ -571,15 +580,25 @@ export function filterMapWithIndex<A, B>(
 export function filterWithIndex<K extends string, A>(fa: Record<K, A>, p: (key: K, a: A) => boolean): Record<string, A>
 export function filterWithIndex<A>(fa: Record<string, A>, p: (key: string, a: A) => boolean): Record<string, A>
 export function filterWithIndex<A>(fa: Record<string, A>, p: (key: string, a: A) => boolean): Record<string, A> {
```

Comment:
What about

```ts
export function filterWithIndex<A>(fa: Record<string, A>, p: (key: string, a: A) => boolean): Record<string, A> {
  const r: Record<string, A> = {}
  let changed = false
  for (const key in fa) {
    if (fa.hasOwnProperty(key)) {
      const a = fa[key]
      if (p(key, a)) {
        r[key] = a
      } else {
        changed = true
      }
    }
  }
  return changed ? r : fa
}
```

Reply:
It's doing all work in case of noop (obj création plus assigns (n log n))

Reply:
noop is unlikely in a `filter*` function so in the most common case looks like you are doing even more work

Reply:
I m fine with the simpler approach.
Afk this morning (on phone) so expect a change later.